Feat zawolanski

zawolanski opened the pull request:
I'm added icons from figma, create Button(only with icon, regarding submit button we have to do it separately) component and i'm applied Poppins font.

zawolanski committed:
add icons

zawolanski committed:
add breakpoints to theme

zawolanski committed:
turn off JSX props spreading eslint rule

zawolanski committed:
add in Poppins font to project

zawolanski committed:
globally reset buttosn and inputs padding

zawolanski committed:
change arrow icon

zawolanski committed:
install hexToRgb package

zawolanski committed:
create button component

pawelsobkowski commented on src/theme/theme.js:
```diff
@@ -11,6 +11,12 @@ const theme = {
     text_blue: '#0E84BF',
     profile_blue: '#003B59',
   },
+  bp: {
```
Clearer name
```suggestion
  breakingPoints: {
```

zawolanski replied:
Ok, i will change it

pawelsobkowski replied:
Thanks

zawolanski committed:
change bp to breakingPoints

zawolanski committed:
set global font size to 62.5%

zawolanski committed:
set border-box to all elements

zawolanski committed:
change dimensions of button component

component is now more flexible and need parent after which he will inherit dimensions

zawolanski committed:
add avatar icon

zawolanski committed:
create icon button component

zawolanski committed:
create header template styles

zawolanski committed:
remove exact from routes

zawolanski committed:
add signin button when user is logged out

zawolanski committed:
remove media queries from IconButton component

pawelsobkowski commented on src/theme/GlobalStyles.js:
```diff
@@ -1,10 +1,23 @@
 const GlobalStyles = createGlobalStyle`
+    :root {
+        font-family: 'Poppins', sans-serif;
+        font-size: 62.5%; 
```
I think that it's better to use a `em` or `rem` unit.

zawolanski replied:
Yes we will use it. It's only reset on html element that let us easier calculate `rem`. Setting it up we get `1rem` = `10px`.

pawelsobkowski commented on src/views/Root/index.js:
```diff
@@ -15,17 +15,17 @@ const Root = () => (
       </Route>
-      <Route exact path={['/', '/home']}>
+      <Route path={['/', '/home']}>
         <Header>
           <Home />
         </Header>
       </Route>
-      <Route exact path="/dashboard">
+      <Route path="/dashboard">
         <Header>
           <Dashboard />
         </Header>
       </Route>
-      <Route exact path="/details/:placeId">
```
Why not `exact` attribute?

zawolanski replied:
If we want to get something like on image belowe we have to let router reach e.g `/home` and `/home/signin` at the same time
![image](https://user-images.githubusercontent.com/61630877/108124372-66b63a80-70a7-11eb-84c0-143845916172.png)

pawelsobkowski replied:
I think that sign in view will be display on foreground, without separate route for this

zawolanski replied:
Ok, but we still have `/search` route that is displaying with `/home` and `/dashboard` route.

zawolanski committed:
position adjusment of icons in header

zawolanski committed:
change :id to :placeId
